src/frequenz/sdk/actor/power_distributing/_component_managers/_pv_inverter_manager/_pv_inverter_manager.py

```diff
@@ -0,0 +1,239 @@
+            .get()
+            .active_power_inclusion_lower_bound,
+            reverse=True,
+        )
+
+        num_components = len(working_components)
+        for idx, inv_id in enumerate(working_components):
+            # Request powers are negative for PV inverters.  When remaining power is
+            # greater than 0.0, we can stop allocating further.
+            if remaining_power > Power.zero() or is_close_to_zero(
+                remaining_power.as_watts()
+            ):
+                break
+            distribution = remaining_power / float(num_components - idx)
+            inv_data = self._component_data_caches[inv_id]
+            if not inv_data.has_value():
+                allocations[inv_id] = Power.zero()
+                # Can't get device bounds, so can't use inverter.
+                continue
+            discharge_bounds = Power.from_watts(
+                inv_data.get().active_power_inclusion_lower_bound
+            )
+            # Because all 3 values are negative or 0, we use max, to get the value
+            # with the least absolute value.
+            allocated_power = max(remaining_power, discharge_bounds, distribution)
+            allocations[inv_id] = allocated_power
+            remaining_power -= allocated_power
+
+        _logger.debug(
+            "Distributing %s to PV inverters %s",
+            request.power,
+            allocations,
+        )
+        await self._set_api_power(request, allocations, remaining_power)
+
+    async def _set_api_power(  # pylint: disable=too-many-locals
+        self, request: Request, allocations: dict[int, Power], remaining_power: Power
+    ) -> None:
+        api_client = connection_manager.get().api_client
+        tasks: dict[int, asyncio.Task[None]] = {}
+        for component_id, power in allocations.items():
+            tasks[component_id] = asyncio.create_task(
+                api_client.set_power(component_id, power.as_watts())
+            )
+        _, pending = await asyncio.wait(
+            tasks.values(),
+            timeout=request.request_timeout.total_seconds(),
+            return_when=asyncio.ALL_COMPLETED,
+        )
+        # collect the timed out tasks and cancel them while keeping the
+        # exceptions, so that they can be processed later.
+        for task in pending:
+            task.cancel()
+        await asyncio.gather(*pending, return_exceptions=True)
+
+        failed_components: set[int] = set()
+        succeeded_components: set[int] = set()
+        failed_power = Power.zero()
+        for component_id, task in tasks.items():
+            exc = task.exception()
+            if exc is not None:
+                failed_components.add(component_id)
+                failed_power += allocations[component_id]
+            else:
+                succeeded_components.add(component_id)
+
+            match task.exception():
+                case asyncio.CancelledError:
```

Comment:
Oh, damn, I just noticed while looking at the new `case`s that this syntax is incorrect. I guess these error cases are not tested, right? `case` must have a "constructor-like" syntax for classes. At least I remember being bitten by this in the past, it is a shame that pylint doesn't seem to warn about this:

```py
>>> import asyncio
>>> e = asyncio.CancelledError("Blah")
>>> match e:
...     case asyncio.CancelledError:
...         print("CE")
...     case _:
...         print("NO")
... 
NO
>>> match e:
...     case asyncio.CancelledError():
...         print("CE")
...     case _:
...         print("NO")
... 
CE
```

> [!WARNING]
> This issue is also present in the other pools!

```suggestion
                case asyncio.CancelledError():
```

Reply:
fixed in all managers